var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -100,30 +111,35 @@ def configure_args(self):
         args.extend(self.enable_or_disable('analysis'))
         args.extend(self.enable_or_disable('lhole'))
         args.extend(self.enable_or_disable('gzip'))
         args.extend(self.enable_or_disable('pythia'))
-        if self.spec.satisfies('+hepmc2'):
-            args.append('--enable-hepmc2=' + self.spec['hepmc'].prefix)
-        if self.spec.satisfies('+hepmc3'):
-            args.append('--enable-hepmc3=' + self.spec['hepmc3'].prefix)
-        if self.spec.satisfies('+rivet'):
-            args.append('--enable-rivet=' + self.spec['rivet'].prefix)
-        if self.spec.satisfies('+fastjet'):
-            args.append('--enable-fastjet=' + self.spec['fastjet'].prefix)
-        if self.spec.satisfies('+openloops'):
-            args.append('--enable-openloops=' + self.spec['openloops'].prefix)
-        if self.spec.satisfies('+recola'):
-            args.append('--enable-recola=' + self.spec['recola'].prefix)
-        if self.spec.satisfies('+root'):
-            args.append('--enable-root=' + self.spec['root'].prefix)
-        if self.spec.satisfies('+lhapdf'):
-            args.append('--enable-lhapdf=' + self.spec['lhapdf'].prefix)
-        if self.spec.satisfies('+hztool'):
-            args.append('--enable-hztool=' + self.spec['hztool'].prefix)
-        if self.spec.satisfies('+cernlib'):
-            args.append('--enable-cernlib=' + self.spec['cernlib'].prefix)
+        hepmc_root = lambda x: self.spec['hepmc'].prefix
+        args.extend(self.enable_or_disable('hepmc2', activation_value=hepmc_root))
+        args.extend(self.enable_or_disable('hepmc3', activation_value='prefix'))
+        args.extend(self.enable_or_disable('rivet', activation_value='prefix'))
+        args.extend(self.enable_or_disable('fastjet', activation_value='prefix'))
+        args.extend(self.enable_or_disable('openloops', activation_value='prefix'))
+        args.extend(self.enable_or_disable('recola', activation_value='prefix'))
+        args.extend(self.enable_or_disable('root', activation_value='prefix'))
+        args.extend(self.enable_or_disable('lhapdf', activation_value='prefix'))
+        # args.extend(self.enable_or_disable('hztool', activation_value='prefix'))
+        # args.extend(self.enable_or_disable('cernlib', activation_value='prefix'))
+        args.extend(self.enable_or_disable('blackhat', activation_value='prefix'))
+        args.extend(self.enable_or_disable('ufo'))
+
+        if self.spec.satisfies('+mpi'):
+            args.append('CC=' + self.spec['mpi'].mpicc)
+            args.append('MPICXX=' + self.spec['mpi'].mpicxx)
+            args.append('CXX=' + self.spec['mpi'].mpicxx)
+            args.append('FC=' + self.spec['mpi'].mpifc)
+
+        if self.spec.satisfies('+cms'):
+            if platform.machine() == 'x86_64':
+                args.append('CXXFLAGS=-fuse-cxa-atexit -m64 -O2 -std=c++0x')
+            else:
+                args.append('CXXFLAGS=-fuse-cxa-atexit -O2 -std=c++0x')
```

Comment:
Instead of adding CXXFLAGS args, you need to use the [Spack flag handlers](https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags) (since users may add flags of their own). Also is `0x` correct? It was superseded by 11.

Reply:
@sethrj I have moved CXXFLAGS to flag handler. I am checking if one can use different `std` value